tensorflow/lite/micro/arc_custom/micro_time.cc

```diff
@@ -25,18 +25,18 @@ limitations under the License.
 
 namespace tflite {
 
-int32_t ticks_per_second() {
+uint32_t ticks_per_second() {
   const unsigned long clocs_per_sec_real = _timer_clocks_per_sec();
   if (clocs_per_sec_real <
-      static_cast<unsigned long>(std::numeric_limits<int32_t>::max()))
-    return static_cast<int32_t>(clocs_per_sec_real);
+      static_cast<unsigned long>(std::numeric_limits<uint32_t>::max()))
+    return static_cast<uint32_t>(clocs_per_sec_real);
   else
-    return std::numeric_limits<int32_t>::max();
+    return std::numeric_limits<uint32_t>::max();
 }
 
-int32_t GetCurrentTimeTicks() {
+uint32_t GetCurrentTimeTicks() {
   const unsigned ticks_real = _timer_default_read();
-  const int32_t ticks_cast = static_cast<int32_t>(ticks_real & 0x7fffffff);
+  const uint32_t ticks_cast = static_cast<uint32_t>(ticks_real & 0x7fffffff);
```

Comment:
I miss this before. There is no need to do & 0x7ffffff when you use uint32_t.  That & is only needed when casting to int32

Reply:
i didn't give it the appropriate amount of consideration - i thought it'd be better to just keep the values, as it might be expected somewhere. looking at some random ARC documentation i found, it seems that there are two counters, one 32b and one 64b.

so, yes, i agree, i should've removed the mask (and will do so now).

i think it does not break anything (other than limiting the length of the time intervals that can be measured, lengths that are probably not relevant for tflite), but it seems pointless and also confusing.

Reply:
pushed the change just now. please let me know whether you would suggest any other changes..